tensorboard/test_util.py

```diff
@@ -0,0 +1,116 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""TensorBoard testing helper routine module.
+
+This module is basically a dumpster for really generic succinct helper
+routines that exist solely for test code.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import threading
+
+
+class FakeClock(object):
+  """Fake version of time.time for testing clock code.
+
+  Classes that call time.time are encouraged to have a clock
+  constructor parameter, which can be swapped out with an instance of
+  this class for testing.
+
+  This class is completely thread safe.
+  """
+
+  def __init__(self, time=1.0):
+    """Creates new instance.
+
+    Args:
+      time: A float of seconds since UNIX epoch in zulu time.
+
+    :type time: float
+    """
+    self._time = time
+    self._lock = threading.Lock()
+
+  def __call__(self):
+    """Returns current fake time.
+
+    Returns:
+      A float of seconds since UNIX epoch in zulu time.
+
+    :rtype: float
+    """
+    with self._lock:
```

Comment:
That all makes sense, so I agree with your call to keep it.

By "tearing" I refer to the phenomenon by which a read of a 64-bit number can result in the lower 32 bits coming from one point in time and the higher 32 bits from another point in time.

As far as I know, the atomic primitives provided by Doug and the JSR-166 team are more targeted toward writes/CAS than reads—indeed, [the OpenJDK implementation of `AtomicInteger#get()`](http://www.grepcode.com/file/repository.grepcode.com/java/root/jdk/openjdk/8u40-b25/java/util/concurrent/atomic/AtomicInteger.java#AtomicInteger.get%28%29) does not lock. (As you pointed out, this doesn't mean that the lock isn't required in Python.)